app/src/test/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileRepoTest.kt

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pinnedtile
+
+import android.arch.lifecycle.Observer
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+/**
+ * Unit tests for the [PinnedTileRepo].
+ * Bundled tile list fetch comes from /assets/bundled/bundled_tiles.json
+ */
+@RunWith(RobolectricTestRunner::class)
+class PinnedTileRepoTest {
+
+    private lateinit var pinnedTileRepo: PinnedTileRepo
+
+    @Before
+    fun setUp() {
+        pinnedTileRepo = PinnedTileRepo(RuntimeEnvironment.application)
+    }
+
+    @Test
+    fun `WHEN repo fetches initial tileList THEN repo should emit a feed of only bundled tile list`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assertEquals(10, (it as LinkedHashMap<String, PinnedTile>).size)
+        })
+
+        pinnedTileRepo.getPinnedTiles().observeForever(observerSpy)
+        pinnedTileRepo.loadTilesCache()
+
+        verify(observerSpy, times(1)).onChanged(any())
+    }
+
+    @Test
+    fun `WHEN repo adds a new pin THEN repo should emit a combined list of bundled and custom`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assert((it as LinkedHashMap<String, PinnedTile>).size == 10 || it.size == 11)
```

Comment:
Could we introduce a `BUNDLED_TILE_COUNT` constant and use in place of these ints?  Otherwise any change to bundled tiles is going to involve a lot of updates to these tests.